lib/spack/spack/compilers/nvhpc.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.compiler import Compiler
+
+
+class Nvhpc(Compiler):
+    # Subclasses use possible names of C compiler
+    cc_names = ['nvc']
+
+    # Subclasses use possible names of C++ compiler
+    cxx_names = ['nvc++']
+
+    # Subclasses use possible names of Fortran 77 compiler
+    f77_names = ['nvfortran']
+
+    # Subclasses use possible names of Fortran 90 compiler
+    fc_names = ['nvfortran']
+
+    # Named wrapper links within build_env_path
+    link_paths = {'cc':  'nvhpc/nvc',
+                  'cxx': 'nvhpc/nvc++',
+                  'f77': 'nvhpc/nvfortran',
+                  'fc':  'nvhpc/nvfortran'}
+
+    PrgEnv = 'PrgEnv-nvhpc'
+    PrgEnv_compiler = 'nvhpc'
+
+    version_argument = '--version'
+    version_regex = r'nv[^ ]* (?:[^ ]+ Dev-r)?([0-9.]+)(?:-[0-9]+)?'
+
+    @property
+    def verbose_flag(self):
+        return "-v"
+
+    @property
+    def debug_flags(self):
+        return ['-g', '-gopt']
+
+    @property
+    def opt_flags(self):
+        return ['-O', '-O0', '-O1', '-O2', '-O3', '-O4']
+
+    @property
+    def openmp_flag(self):
+        return "-mp"
```

Comment:
If we don't need to compute the string based on properties, can't we just do:
```suggestion
    openmp_flag = "-mp"
```
for all the properties?

Reply:
It could.  Currently it's just following the convention used by the most of the compilers.  

For `openmp_flag` in particular:
- `apple_clang` and `clang` directly set `openmp_flag`
- `cce` and `intel` depend on the object state
- the rest use the `@property` decorated getter without depending on the object state.

Reply:
I'm fine with either syntax. Technically they aren't identical, as a getter without a setter won't let you set, but I don't think we've ever needed that functionality.

Reply:
> Technically they aren't identical, as a getter without a setter won't let you set

I lean toward keeping it as is for this reason, but I'm happy to change it if @alalazo feels strongly about it.